std/json.d

```diff
@@ -707,22 +707,36 @@ if (isInputRange!T && !isInfinite!T && isSomeChar!(ElementEncodingType!T))
     JSONValue root;
     root.type_tag = JSON_TYPE.NULL;
 
+    // UTF decoding is unnecessary when parsing JSON.
+    static if (is(T : const(char)[]))
+        alias Char = char;
```

Comment:
This breaks stuff:

```d
parseJSON("\"\U0001D11E\"");
    /* std.json.JSONException@std/json.d(1374): Illegal control character. (Line 1:3) */
```

The thing is that the rest of the code assumes `dchar`. While `char` converts to `dchar`, it doesn't always keep the same meaning. So stuff gets misinterpreted.

In this specific case, `isControl` is given a `char` that becomes a control character when converted to `dchar`. So it rings the alarm even though the encoded code point in the string is not a control character. Could also go the other way: When a control character is encoded with `char`s that don't look like control characters, it slips through (haven't checked if this can actually happen).